docs/topics/gradle/gradle-configure.md

```diff
@@ -0,0 +1,745 @@
+}
+```
+
+</tab>
+</tabs>
+
+When configuring your project, check the Kotlin Gradle plugin compatibility with available Gradle versions:
+
+|                       | Minimum supported version | Maximum fully supported version |
+|-----------------------|---------------------------|---------------------------------|
+| Gradle                | %minGradleVersion%        | %maxGradleVersion%              |   
+| Android Gradle plugin | %minAndroidGradleVersion% | %maxAndroidGradleVersion%       |
+
+For example, the Kotlin Gradle plugin and the `kotlin-multiplatform` plugin %kotlinVersion% require the minimum Gradle
+version of %minGradleVersion% for your project to compile.
+
+In turn, the maximum fully supported version is %maxGradleVersion%. It doesn't have deprecated Gradle
+methods and properties and supports all the current Gradle features.
+
+## Targeting the JVM
+
+To target the JVM, apply the Kotlin JVM plugin.
+
+<tabs group="build-script">
+    <tab title="Kotlin" group-key="kotlin">
+
+```kotlin
+plugins {
+    kotlin("jvm") version "%kotlinVersion%"
+}
+```
+
+</tab>
+<tab title="Groovy" group-key="groovy">
+
+```groovy
+plugins {
+    id "org.jetbrains.kotlin.jvm" version "%kotlinVersion%"
+}
+```
+
+</tab>
+</tabs>
+
+The `version` should be literal in this block, and it cannot be applied from another build script.
+
+Alternatively, you can use the older `apply plugin` approach:
+
+```groovy
+apply plugin: 'kotlin'
+```
+
+Applying Kotlin plugins with `apply` in the Kotlin Gradle DSL is not recommended – [see why](gradle-kotlin-dsl.md#using-the-gradle-kotlin-dsl).
+
+### Kotlin and Java sources
+
+Kotlin sources and Java sources can be stored in the same folder, or they can be placed in different folders. The default convention is to use different folders:
+
+```groovy
+project
+    - src
+        - main (root)
+            - kotlin
+            - java
+```
+
+The corresponding `sourceSets` property should be updated if you are not using the default convention:
+
+<tabs group="build-script">
+<tab title="Kotlin" group-key="kotlin">
+
+```kotlin
+sourceSets.main {
+    java.srcDirs("src/main/myJava", "src/main/myKotlin")
+}
+```
+
+</tab>
+<tab title="Groovy" group-key="groovy">
+
+```groovy
+sourceSets {
+    main.kotlin.srcDirs += 'src/main/myKotlin'
+    main.java.srcDirs += 'src/main/myJava'
+}
+```
+
+</tab>
+</tabs>
+
+### Check for JVM target compatibility of related compile tasks
+
+In the build module, you may have related compile tasks, for example:
+* `compileKotlin` and `compileJava`
+* `compileTestKotlin` and `compileTestJava`
+
+> `main` and `test` source set compile tasks are not related.
+>
+{type="note"}
+
+For such related tasks, the Kotlin Gradle plugin checks for JVM target compatibility. Different values of `jvmTarget` in
+the `kotlin` extension and [`targetCompatibility`](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java-extension)
+in the `java` extension cause incompatibility. For example:
```

Comment:
```suggestion
in the `java` extension can cause incompatibility issues. For example:
```

Reply:
We try to avoid words like issues, problems, etc.

Reply:
Do you mean there should be some specifying word before "incompatibility"? Because above we write:
> the Kotlin Gradle plugin checks for JVM target compatibility

And then we write that some conditions may cause incompatibility. Incompatibility is quite a widespread word in development. 


In the chapter about [inline functions](https://kotlinlang.org/docs/inline-functions.html#restrictions-for-public-api-inline-functions), we write:

> This imposes certain risks of binary incompatibility caused by changes in the module that declares an inline function in
> case the calling module is not re-compiled after the change.
> 
> To eliminate the risk of such incompatibility being introduced by a change in a *non*-public API of a module, public
> API inline functions are not allowed to use non-public-API declarations, i.e. `private` and `internal` declarations and
> their parts, in their bodies.

But there we use "binary" and "such".

Reply:
So "JVM target compatibility" is the same as saying "compatibility of the JVM targets". Similarly with "binary". And "such" is just a shortcut meaning "binary" again. So two different JVM target versions in different parts of the build system can be incompatible. To say "causes incompatibility" we need to add "of something". It might just be easier to say that the JVM targets must be the same 😃

Reply:
>two different JVM target versions in different parts of the build system 
Not the build system, but the build settings. Gradle is the build system. All other stuff is the means with which we use Gradle. But it doesn't matter.

Let' say 
cause JVM target incompatibility

I suppose it's better to show the example with differences as it's now than say that the targets should be the same.